bin/test_integration

```diff
@@ -2,13 +2,46 @@
 
 set -eo pipefail
 
+help_and_exit() {
+  local retval=${1:-1}
+  cat <<EOF
+${0##*/} [-l|--local] [test_folder1] [test_folder2] [test_folder3] ...
+
+Runs suite of integration tests by calling ./start and ./stop scripts, if found, for subdirectories in the test subdirectory of the project root.
+Names of subdirectories of the project root can be optionally specified as arguments to limit the subdirectories whose test cases will be run.
+
+Options:
+  -l, --local sets the local flag when the test is called in the test folder
+EOF
+  exit "$retval"
+}
+
+non_flag_args=( )
+test_flag="";
+while (( $# )); do
+  case $1 in
+    -h|--help)   help_and_exit 0 ;;
+    -l|--local)  test_flag="-l" ;;
+    -*)          echo "Unknown option: $1";
+                 echo "";
+                 help_and_exit 1 ;;
+    *)           non_flag_args+=( "$1" ) ;;
+  esac
+  shift
+done
+
+test_folders=*/;
+if [[ ${#non_flag_args[@]} -gt 0 ]]; then
+  test_folders=${non_flag_args}
```

Comment:
See https://github.com/koalaman/shellcheck/wiki/SC2128

Again, this sort of works by accident now because the array only _has_ one element (see comment above), but better to use arrays properly here.